ipatests/test_integration/test_commands.py

```diff
@@ -1646,6 +1646,89 @@ def test_sid_generation(self):
+        server context. This test checks that groups have correct userclass
+        when external is set to true or false with group-add.
+
+        related: https://pagure.io/freeipa/issue/9349
+        """
+        user_code_script = textwrap.dedent("""
+            from ipalib import api, errors
+            api.bootstrap_with_global_options(context='server')
+            api.finalize()
+            if api.env.in_server:
+                api.Backend.ldap2.connect()
+            else:
+                api.Backend.rpcclient.connect()
+            try:
+                api.Command["group_del"]("testgroup1")
+            except errors.NotFound:
+                pass
+            try:
+                api.Command["group_del"]("testgroup2")
+            except errors.NotFound:
+                pass
+            api.Command["group_add"]("testgroup1", external=True)
+            api.Command["group_add"]("testgroup2", external=False)
+            result1 = api.Command["group_show"]("testgroup1", all=True)["result"] # noqa: E501
+            result2 = api.Command["group_show"]("testgroup2", all=True)["result"] # noqa: E501
+            print("'testgroup2' userclass: %s" % repr(result2["objectclass"]))
+        """)
+        self.master.put_file_contents("/tmp/reproducer1_code.py",
+                                      user_code_script)
+        result = self.master.run_command(['python3',
+                                          '/tmp/reproducer1_code.py'])
+        assert "ipaexternalgroup" not in result.stdout_text
+
+    def test_sequence_processing_nonposix_group(self):
+        """Test for sequence processing failures
+
+        Issues have been found for group_add sequence processing with
+        server context after creating a nonposix group. This test checks
+        that all following group_add calls to add posix groups calls are
+        not failing with missing attribute.
+
+        related: https://pagure.io/freeipa/issue/9349
+        """
+        user_code_script2 = textwrap.dedent("""
+            from ipalib import api, errors
+            api.bootstrap_with_global_options(context='server')
+            api.finalize()
+            if api.env.in_server:
+                api.Backend.ldap2.connect()
+            else:
+                api.Backend.rpcclient.connect()
+            try:
+                api.Command["group_del"]("testgroup1")
+            except errors.NotFound:
+                pass
+            try:
+                api.Command["group_del"]("testgroup2")
+            except errors.NotFound:
+                pass
+            try:
+                api.Command["group_del"]("testgroup3")
+            except errors.NotFound:
+                pass
+            api.Command["group_add"]("testgroup1", nonposix=False)
+            try:
+                api.Command["group_add"]("testgroup2", nonposix=True)
+            except Exception as e:
+                print(str(e))
```

Comment:
I'd use this instead (and for testgroup3): `print("testgroup2: %s" % e)`
This way it is more visible which group failed and that you caught it.

It's hard to tell what you're expecting here. I think just testgroup2 will fail, right? So the catch in testgroup3 is misleading because you aren't actively looking at the output.